### PR TITLE
[CI] Enable build stage for arm architecture

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use pipeline 2.0 to build in arm

## Why is it important?

Enable arm64 builds/tests when merging to branches or tags creation.

## When
- By default on builds from branches.
- By default on builds from tags.
- For PRs:
  - if label `arm` is added then the following builds will run it.
  - if comment `/test <module> for arm"` it will trigger a build for arm. <module> matches the beat folder name or x-pack/<beat-folder-name>. For instance `/test x-pack/auditbeat for arm` or `/test auditbeat for arm`

## Notes

- Uses [arm](https://beats-ci.elastic.co/label/arm/) label
- Uses https://github.com/elastic/beats/pull/20104 as the base branch. (That's the reason for all those commits)
- https://github.com/elastic/beats/pull/21284/commits/d7339674fc86ee48be56de1482698af34f726d5c are the potential changes to enable arm builds in auditbeat.
- Support gvm/go installation in arm architecture.

## Furthermore

Directory | OS | Mage Target | Status CI |
-- | -- | -- | -- |
auditbeat | linux/arm64 | build unitTest | ✅ 
filebeat | linux/arm64 | build unitTest | ✅ 
heartbeat | linux/arm64 | build unitTest | ✅ 
journalbeat | linux/arm64 | build unitTest | ✅ 
libbeat | linux/arm64 | build unitTest | ✅ 
metricbeat | linux/arm64 | build unitTest | 🔴 
packetbeat | linux/arm64 | build unitTest | ✅ 
x-pack/auditbeat | linux/arm64 | build unitTest | ✅ 
x-pack/elastic-agent | linux/arm64 | build unitTest | ✅ 
x-pack/filebeat | linux/arm64 | build unitTest | ✅ 
x-pack/libbeat | linux/arm64 | build unitTest | ✅ 
x-pack/metricbeat | linux/arm64 | build unitTest | 🔴 
x-pack/functionbeat | linux/arm64 | build unitTest | ✅ 

## Issues

Test failures in metricbeats and x-pack/metricbeats:
- https://github.com/elastic/beats/issues/21304
- https://github.com/elastic/beats/issues/21301

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

**Review by mdelapenya — APPROVED** [review]
Love the readability of the DSL!

File: Jenkinsfile [hunk 1]
```diff
@@ -46,6 +45,7 @@ pipeline {
     booleanParam(name: 'awsCloudTests', defaultValue: true, description: 'Run AWS cloud integration tests.')
     string(name: 'awsRegion', defaultValue: 'eu-central-1', description: 'Default AWS region to use for testing.')
     booleanParam(name: 'runAllStages', defaultValue: false, description: 'Allow to run all stages.')
+    booleanParam(name: 'armTest', defaultValue: false, description: 'Allow arm stages.')
```
Comment (cachedout) [inline Jenkinsfile:48]:
```suggestion
    booleanParam(name: 'armTest', defaultValue: false, description: 'Allow ARM stages.')
```

**Review by cachedout — APPROVED** [review]

**Comment by v1v** [comment]
Environmental issue with `Build&Test / libbeat-build`

```bash
$ mage build test
...
[2020-10-14T15:42:03.951Z] Building elasticsearch_kerberos.elastic
[2020-10-14T15:42:12.178Z] Service 'elasticsearch_kerberos.elastic' failed to build: The command '/bin/sh -c /scripts/installkdc.sh && /scripts/addprincs.sh' returned a non-zero code: 127
[2020-10-14T15:42:12.178Z] >> Building docker images again
[2020-10-14T15:42:12.178Z] elasticsearch uses an image, skipping
[2020-10-14T15:42:12.178Z] elasticsearch_monitoring uses an image, skipping
[2020-10-14T15:42:12.178Z] elasticsearchssl uses an image, skipping
[2020-10-14T15:42:12.178Z] logstash uses an image, skipping
[2020-10-14T15:42:12.178Z] Building redis
[2020-10-14T15:42:13.124Z] Building sredis
[2020-10-14T15:42:14.070Z] Building kafka
[2020-10-14T15:42:14.644Z] Building elasticsearch_kerberos.elastic
[2020-10-14T15:42:16.047Z] Service 'elasticsearch_kerberos.elastic' failed to build: The command '/bin/sh -c /scripts/installkdc.sh && /scripts/addprincs.sh' returned a non-zero code: 127
[2020-10-14T15:42:16.047Z] Error: running "docker-compose -p libbeat_8_0_0_b2fbc1b280-snapshot build --force-rm --pull" failed with exit code 1
script returned exit code 1[2020-10-14T15:42:16.417Z] Client: Docker Engine - Community
```

**Pull request merged**